eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -301,6 +301,30 @@ jobs:
       coverageThreshold: 70
     condition: eq(variables['AZ_SDK_CODE_COV'], 1)
 
+  # Validate all the files are saved as ASCII only without a UTF-8 BOM.
+  - bash: |
+      shopt -s globstar
+
+      files_with_non_ascii=0
+      for file in . ./* ./**/* ; do
+        if [ ! -d "$file" ]; then
+          echo "Validating that $file file only contains ASCII bytes..."
+          if grep -I --color='auto' -P -n "[^\x00-\x7F]" $file ; then
+            if [[ $file == *"CodeCoverage.cmake"* ]]; then
+              echo "Skipping errors from $file since it contains a name."
+            else
+              ((files_with_non_ascii++))
+            fi
+          fi
+        fi
+      done
+
+      echo Files found with non-ASCII characters: $files_with_non_ascii
+      exit $files_with_non_ascii
+      
+    displayName: Validate Characters are ASCII
+    condition: eq(variables['AZ_SDK_CODE_COV'], 1)
```

Comment:
This condition ensures that the step only runs once among our matrix. The `GenerateReleaseArtifacts` job only runs once per-pipeline-execution and might be a better place for this. 

Perhaps the job should be renamed to `AnalyzeAndGenerateReleaseArtifacts`

Reply:
I didn't want to run this on a windows VM since it relies on grep and bash/linux-only commands.

Reply:
Good point. We do have bash on the Windows VM but I don't want to tempt unknown behaviors of running bash on windows. This is a good place for it for the time being.